gt/lua_lpm.c

```diff
@@ -491,29 +507,31 @@ static const struct luaL_reg lpmlib_lua_c_funcs [] = {
 };
 
 static int
-lpm_gc(lua_State *l) {
-	struct rte_lpm *lpm = *(struct rte_lpm **)lua_touserdata(l, 1);
-	rte_lpm_free(lpm);
+lpm_ud_gc(lua_State *l) {
+	struct lpm_lua_userdata *lpm_ud =
+		(struct lpm_lua_userdata *)lua_touserdata(l, 1);
```

Comment:
Can't we drop the typecase here as well?

This comment applies to the IPv6 case below as well.